src/vstest.console/CommandLine/CommandLineOptions.cs

```diff
@@ -282,13 +283,29 @@ internal Framework TargetFrameworkVersion
         /// <param name="source">Path to source file to look for tests in.</param>
         public void AddSource(string source)
         {
+            var extnList = new List<string>() { ".xap", ".appx", ".dll", ".exe" };
             if (String.IsNullOrWhiteSpace(source))
             {
                 throw new CommandLineException(CommandLineResources.CannotBeNullOrEmpty);
             }
-            
+
             source = source.Trim();
-            
+
+            //verify whether the given argument is source type if not process further
+            int count = 0;
+            foreach (string extension in extnList)
```

Comment:
read through [https://stackoverflow.com/questions/36138351/how-to-write-a-linq-statement-to-get-files-with-specific-extensions-from-a-dynam](https://stackoverflow.com/questions/36138351/how-to-write-a-linq-statement-to-get-files-with-specific-extensions-from-a-dynam)

Reply:
Thank you for the link, implemented and pushed it